src/App/src/Middleware/DeprecationMiddleware.php

```diff
@@ -88,41 +173,20 @@ public function process(
                 )
             );
         }
-
-        $sunset = null;
-        $link   = null;
-        if (array_key_exists(self::RESOURCE_DEPRECATION_ATTRIBUTE, $attributes)) {
-            $sunset = $attributes[self::RESOURCE_DEPRECATION_ATTRIBUTE]['sunset'];
-            $link   = $attributes[self::RESOURCE_DEPRECATION_ATTRIBUTE]['link'];
-        }
-
-        if (array_key_exists(self::METHOD_DEPRECATION_ATTRIBUTE, $attributes)) {
-            $sunset = $attributes[self::METHOD_DEPRECATION_ATTRIBUTE]['sunset'];
-            $link   = $attributes[self::METHOD_DEPRECATION_ATTRIBUTE]['link'];
-        }
-
-        if ($sunset !== null) {
-            $response = $response->withHeader('sunset', $sunset);
-        }
-
-        if ($link !== null) {
-            $response = $response->withHeader('link', $link);
-        }
-
-        return $response;
     }
 
-    /**
-     * @param class-string $type
-     */
-    public function getAttributes(ReflectionClass|ReflectionMethod $reflection, string $type): array
+    private function formatLink(string $baseLink, array $attribute): string
     {
-        $attributes = [];
-        foreach ($reflection->getAttributes($type) as $attribute) {
-            $attribute->newInstance();
-            $attributes[$attribute->getName()] = $attribute->getArguments();
+        $parts = [
+            $baseLink,
+        ];
+        if (! empty($attribute['rel'])) {
+            $parts[] = sprintf('rel="%s"', $attribute['rel']);
+        }
+        if (! empty($attribute['type'])) {
+            $parts[] = sprintf('type="%s"', $attribute['type']);
         }
 
-        return $attributes;
+        return rtrim(implode(';', $parts), ';');
```

Comment:
```suggestion
        return implode(';', $parts);
```
No need for rtrim because imploding by semicolon will only add them between the parts - there will be no semicolon at the end of the string.